src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1482,6 +1482,13 @@ private string GetLocalDebugName(ILocalSymbolInternal local, out LocalDebugId lo
                 return null;
             }
 
+            // Skip naming locals generated by instrumentation as they might not have a local syntax offset.
+            // Locals generated by instrumentation might exist in methods which do not contain a body (auto property initializers).
+            if (local.SynthesizedKind == SynthesizedLocalKind.InstrumentationPayload)
+            {
+                return null;
```

Comment:
I forget precisely where, but without the change there is an assertion failure because the presence of a local variable that is visible to the debugger is assumed to imply the existence of a syntactic block, and there is no block for an autoprop.

There is no need to make these synthesized locals visible to the debugger, so keeping them invisible seems sound.